server_development/topics/admin-rest-api.adoc

```diff
@@ -37,6 +39,29 @@ curl \
   "http://localhost:8080/auth/admin/realms/master"
 ----
 
+==== Authenticate with a service account
+
+Before being able to authenticate against the Admin REST API using a `client_id` and a `client_secret` you need to make sure the client is configured as it follows:
+
+* `client_id` is a **confidential** client that belongs to the realm *master*
+* `client_id` has `Service Accounts Enabled` option enabled
+* `client_id` has a custom "Hardcoded claim" mapper
+  ** Token Claim Name: `azp` 
```

Comment:
Setting the azp here is not valid, and is misleading. The client should have scope on the required roles within security-admin-console, limited to what it needs. Then it should have the 'audience resolve' mapper added so the token has an audience on security-admin-console. which will be added automatically as it as a scope on the roles for that client.

Reply:
Hi Stian,

thanks for your comment. I will fix it.

>  The client should have scope on the required roles within security-admin-console, limited to what it needs
Can you clarify what you mean by this? I am not sure how I can translate that sentence into keycloak config (I am trying to reproduce what you say so I can eventually document it in plain english)

Reply:
The steps should create a new client (not re-use admin-client client), remove full scope allowed, and select what roles it needs from the security-admin-console, then it should have the audience resolve mapper added. That will mean it has access to the roles it needs + the correct audience.

Reply:
Thanks for your comments. I just updated my PR.